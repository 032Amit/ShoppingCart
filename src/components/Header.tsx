import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

interface IHeaderProps {
  title: string;
}
const Header = (props: IHeaderProps) => {
  const {title} = props;
  return (
    <View style={styles.container}>
      <Text style={styles.title}>{title}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: 60,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    paddingLeft: 20,
    backgroundColor: '#fff',
    elevation: 1,
  },
  title: {
    color: '#000',
    fontSize: 16,
    fontWeight: '700',
  },
});

export default Header;
