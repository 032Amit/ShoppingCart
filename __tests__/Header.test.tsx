import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../src/components/Header';

test('renders header with title', () => {
  const title = 'Flipkart';
  const component = renderer.create(<Header title={title} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('updates header title', () => {
  const component = renderer.create(<Header title="Amazon" />);
  const newTitle = 'PRETTYLITTLETHING';

  component.update(<Header title={newTitle} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('handles empty title', () => {
  const component = renderer.create(<Header title="" />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
