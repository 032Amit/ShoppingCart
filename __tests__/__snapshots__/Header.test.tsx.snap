// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles empty title 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "elevation": 1,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "paddingLeft": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#000",
        "fontSize": 16,
        "fontWeight": "700",
      }
    }
  />
</View>
`;

exports[`renders header with title 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "elevation": 1,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "paddingLeft": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#000",
        "fontSize": 16,
        "fontWeight": "700",
      }
    }
  >
    Flipkart
  </Text>
</View>
`;

exports[`updates header title 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "elevation": 1,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "paddingLeft": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#000",
        "fontSize": 16,
        "fontWeight": "700",
      }
    }
  >
    PRETTYLITTLETHING
  </Text>
</View>
`;
